lib/vfscore/syscalls.c

```diff
@@ -859,63 +859,48 @@ sys_rename(char *src, char *dest)
 int
 sys_symlink(const char *oldpath, const char *newpath)
 {
-	struct task	*t = main_task;
-	int		error;
-	char		op[PATH_MAX];
-	char		np[PATH_MAX];
-	struct dentry	*newdp;
-	struct dentry	*newdirdp;
-	char		*name;
-
-	if (oldpath == NULL || newpath == NULL) {
-		return (EFAULT);
-	}
+	int error;
+	char np[PATH_MAX];
+	struct dentry *newdp = NULL;
+	struct dentry *newdirdp = NULL;
+	char *name;
```

Comment:
Again, maybe just my preference, but now that there seems to be some sort of refactoring process on-going I guess, I think everything would look more beautiful if local variable declarations would be done in a reverse xmas tree order w.r.t. declaration line length. But just my preference. Another very helpful thing, I believe, would be adding the include files in alphabetical order, as that seems to be the consensus on some other large OSS projects, though this is not necessarily related to this PR :).

Reply:
Yes, I like reverse xmas also better. Changed it.